nx/lib/nx/lin_alg.ex

```diff
@@ -348,7 +348,7 @@ defmodule Nx.LinAlg do
   end
 
   defp norm_integer(%{shape: {_, _}} = t, -2, _opts) do
-    {_u, s, _v} = svd(t)
+    {_u, s, _v} = Nx.LinAlg.svd(t)
```

Comment:
@josevalim I'm not sure what's happening, but this call to SVD is trying to call the BinaryBackend directly instead of the default implementation provided

Reply:
Is it happening inside `defn` or outside? It may be related to the fact `norm_integer` is not inside a `defn`?

Reply:
It happens on `norm`'s doctests. Might have something to do with the fact that norm is a defn which calls norm_transform, which then calls the svd defn.

Given that case is now supported, we can probably ditch the transform. I'll try that to see if the errors go away.

Reply:
They didn't go away. As the code became a bit less readable due to no-multiclause and no-guards constraints for defns, I kept it as is for now